fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function fetchUserID() {
+      try {
+        const response = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`,
+          { method: 'GET' }
+        );
```

Comment:
Whenever you add dynamic data to a URL (such as `userId`), make sure to encode it with `encodeURIComponent` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) so it doesn't accidentally break the URL. It may not matter here if `userId` is always a number, but it's a good habit to get into.